maintenance/measure_tile_performance/measure_tile_performance.py

```diff
@@ -15,24 +16,59 @@
 NAME = 'snowplow'
 HOST = 'redshift.analytics.gov.bc.ca'
 PORT = 5439
+user = os.environ['lookeruser_rs']
+passwd = os.environ['lookerpass_rs']
+connection_string = (
+    f"dbname='{NAME}' "
+    f"host='{HOST}' "
+    f"port='{PORT}' "
+    f"user='{user}' "
+    f"password={passwd}")
 
 # connects with redshift database using environment variables
 # set redshift cache to 'off' before running any looker API calls
 # run the API calls
 # set redshift cache back to 'on' after the queries are processed.
 # close the redshift connection
+
+def cache_off():
```

Comment:
significant overlap in code between `cache_off()` and `cache_on()`. Is it necessary? If changes to the psycopg2 library necessitate changes to how it connects or executes, we would now need to update this script in two locations.

Why not just have a single `cache(status)` function that accepts a string argument, _`status`_, as `on` or `off` and otherwise fails (or defaults to `on` for safety)?

you can get the important work done by something like:

```python
try:
  curs.execute(f"ALTER USER looker SET enable_result_cache_for_session TO {status};")
  print(f'redshift cache is {status} and connection is closed')
except Exception as err:
  print(f'Exiting due to psycopg2 execution error: {err}')
  print('redshift cache is {'on' if x == 'off' else 'off'} and connection is closed')
  exit(1)
```

You may want a line or two more logic between if for some reason you don't want to attempt the repeat attempts if the cache is being set off; up to you.

Reply:
Good idea, Let me try merge both functions.

Reply:
Merged both functions in the latest commit